### PR TITLE
Feat: add all icons from figma to project

### PR DESCRIPTION
**Pull request by zoheirkhonyagar**
all of the icons have been imported from Figma to the project

### CODE REVIEW

File: components/Button.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const Button: FC<ButtonProps> = ({ type, key, children }) => {
       className={`${colorVariants[type]} flex content-center items-center justify-center space-x-1.5 rounded-md px-1`}
     >
       {children}
-      <p className="flex">{type}</p>
+      {type !== 'Hug' ? <p className="flex">{type}</p> : null}
```
Comment (maryamrmz) [inline components/Button.tsx]: Why we use `a` tag in the component that its name is `Button`?
also what do you think about this:
`{type !== 'Hug' && <p className="flex">{type}</p>}`

Comment (zoheirkhonyagar) [reply components/Button.tsx]: it's just temporary. If you have a better suggestion please replace it with a tag that is more relevant,
 the whole MR is about just adding svg icons from figma to the project.

File: components/Icon/icon.type.ts [hunk 2]
```diff
@@ -1,33 +1,24 @@
...
 }
 
-export interface SVGIcon {
-  path: SVG[];
-  options?: SVGOption;
-}
+export type Icon = (props: IconProps) => JSX.Element;
 
-export type iconNames = 'star' | 'bold-star' | 'search-normal';
+export type iconNames =
```
Comment (maryamrmz) [inline components/Icon/icon.type.ts]: Type names should start with a capital letter to keep the code consistent.

File: components/Icon/icon.type.ts [hunk 3]
```diff
@@ -1,33 +1,24 @@
...
-  strokeWidth?: string;
-  strokeMiterlimit?: string;
-}
+import { SVGProps } from 'react';
 
-export interface SVGOption {
-  viewBox?: string;
+// Create interface extending SVGProps
+export interface IconProps extends Partial<SVGProps<SVGSVGElement>> {
+  size?: string | number;
```
Comment (maryamrmz) [inline components/Icon/icon.type.ts:5]: Why `string` or `number`?

Comment (zoheirkhonyagar) [reply components/Icon/icon.type.ts:5]: this interface has been copied from: https://github.com/shadcn/ui
I didn't change the type that they used it there. there might be a reason for it but I didn't check it TBH.

File: components/index.ts [hunk 4]
```diff
@@ -1,3 +1,3 @@
 export { default as Button } from './Button';
 export { default as Card } from './Card';
-export { default as Icon } from './Icon';
+export { Icons } from './Icon';
```
Comment (maryamrmz) [inline components/index.ts]: I think this line has violated consistency. Is there any specific reason for that?

Comment (zoheirkhonyagar) [reply components/index.ts]: on it, I'm gonna fix it. thanks for noticing it.

File: components/Icon/icon.type.ts (hunk 3 above)
Comment (maryamrmz) [reply components/Icon/icon.type.ts:5]: I can't find that! Can you link to the same file?

Comment (zoheirkhonyagar) [reply components/Icon/icon.type.ts:5]: image
it's inside a definition file inside the node_modules.

**Review by maryamrmz — APPROVED** [review]

**Pull request merged**